src/api/client.js

```diff
@@ -18,10 +25,31 @@ function assetTypesFromState(assetTypes) {
   return Object.keys(assetTypes).filter(key => assetTypes[key]);
 }
 
-export function requestAssets(courseId, { page = 0, pageSize = 50, sort = 'sort', assetTypes = {} }) {
-  const assetTypesToFilter = assetTypesFromState(assetTypes);
+function getDatabaseAttributesFromAssetAttributes(sort) {
+  return assetAttributesToDatabaseAttributes[sort];
+}
+
+function objectIsNonEmpty(object) {
+  return Object.keys(object).length > 0;
+}
+
+export function requestAssets(courseId, params) {
+  const assetTypesToFilter = assetTypesFromState(params.assetTypes);
+  const sortType = getDatabaseAttributesFromAssetAttributes(params.sort);
+
+  const parameters = {
+    ...params,
+    sort: sortType,
+    asset_type: (objectIsNonEmpty(assetTypesToFilter) ? assetTypesToFilter : undefined),
```

Comment:
Sigh. The value of assetTypes in Redux's parameters state is {}. The reduce function at line 49 ends up tacking on the substring "assetTypes=", since parameters[assetType] is true, given that assetType = {}. Therefore, I had to basically add the assertion that we only add assetTypes to the request string if parameters[key].legth > 0 (i.e. assetTypes exists but isn't just {}). Should we just change the default initialState in the Redux store to undefined?

Reply:
Will it break anything if we do set it to `undefined` in the redux store instead of `{}`? If not, I'm all for uniformity.  If yes, I think this is fine for now.